src/betterproto/__init__.py

```diff
@@ -693,8 +693,21 @@ def __repr__(self) -> str:
         def __getattribute__(self, name: str) -> Any:
             """
             Lazily initialize default values to avoid infinite recursion for recursive
-            message types
+            message types.
+            Raise :class:`AttributeError` on attempts to access unset ``oneof`` fields.
             """
+            try:
+                group_current = super().__getattribute__("_group_current")
+            except AttributeError:
+                pass
+            else:
+                if name not in {"__class__", "_betterproto"}:
+                    group = self._betterproto.oneof_group_by_field.get(name)
+                    if group is not None and group_current[group] != name:
+                        raise AttributeError(
+                            f"'{self.__class__.__name__}.{group}' is set to '{group_current[group]}', not '{name}'"
+                        )
```

Comment:
```suggestion
                        if sys.version_info < (3, 10):
                            raise AttributeError(
                                f"{group!r} is set to {group_current[group]}, not {name!r}"
                            )
                        else:
                            raise AttributeError(
                                f"{group!r} is set to {group_current[group]}, not {name!r}",
                                name=name,
                                obj=self,
                            )
```
Should make error messages nicer if python/cpython#103232 is ever merged

Reply:
I've updated the change.